app/views/devise/registrations/edit.html.haml

```diff
@@ -2,9 +2,14 @@
 = simple_form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put, class: "form-horizontal" }) do |f|
   = f.error_notification
   .form-inputs
-    = f.input :email, required: true, autofocus: true
-    = f.input :password, autocomplete: "off", hint: t("devise.registrations.leave_it_blank_if_dont_want_to_change"), required: false
-    = f.input :password_confirmation, required: false
-    = f.input :current_password, hint: t("devise.registrations.need_current_password_to_confirm"), required: true
-  .form-actions
-    = f.button :submit, t('common.update'), class: "btn btn-primary"
+    - if current_user.demo_user?
+      = f.input :email, required: false, autofocus: false, :input_html => { readonly: :true, :value => 'demo@homebugh.info' }
+      = f.input :password, autocomplete: "off", hint: t("devise.registrations.demo_user_account_defaults"), required: false, :input_html => { readonly: :true, :value => 'demo1234' }
```

Comment:
```suggestion
      = f.input :password, autocomplete: "off", hint: t("devise.registrations.demo_user_account_defaults"), required: false, input_html: { readonly: true, value: 'demouser', type: 'text' }
```

Here I've suggested the new Ruby hash syntax as well. Also, I think we need to add `type: 'text'` here, otherwise the password is going to be hidden from users. I've also changed it to "demouser", since I've created a user with that password already.

Do we need a `hint` here? What was your idea behind it? What message do you think we need to use there?

Reply:
Awesome!! Your changes look good to me ... my idea with the hint was to show some text/output for anyone who uses the demo_user and wonders why can't change the inputs but I guess the account speak by itself. 

I will delete the hint to make the code looked tidier if you are happy with it